### PR TITLE
Add option to switch between KG and LBS

### PR DESCRIPTION
**Pull request by MisterChocker**
## Summary of Changes

- Added Option to MCDU
- Adjusted Upper ECAM
- Adjusted Lower ECAM
- Adjusted Fuel Page
- Adjusted INIT-B
- Adjusted FUEL PRED PAGE

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by Benjozork** [comment]
Some PRs might be blocking this

**Comment by MisterChocker** [comment]
@Benjozork Which PRs unparticular?

**Comment by Benjozork** [comment]
#2251 is the only one I can see rn

**Comment by MisterChocker** [comment]
Then we need to get the pr tested ASAP

**Comment by wpine215** [comment]
> 
> 
> Then we need to get the pr tested ASAP

#2251 looked good to me, so I merged.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_FuelPredPage.js [hunk 1]
```diff
@@ -134,8 +134,8 @@ class CDUFuelPredPage {
                         rteRsvPercentCell = "--.-";
                         rteRSvCellColor = "[color]cyan";
                     } else {
-                        rteRsvWeightCell = mcdu.getRouteReservedWeight().toFixed(1);
-                        rteRsvPercentCell = mcdu.getRouteReservedPercent().toFixed(1);
+                        rteRsvWeightCell = (mcdu.getRouteReservedWeight() * mcdu._conversionWeight).toFixed(1);
+                        rteRsvPercentCell = (mcdu.getRouteReservedPercent() * mcdu._conversionWeight).toFixed(1);
```
Comment (Lucky38i) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_FuelPredPage.js:138]: This is a percentage not a weight

**Review by Lucky38i — CHANGES_REQUESTED** [review]
You need to amend all the range checks as well otherwise people who are using lbs will get a lot of "entry out of range" checks.

Maybe something like..
`return this.conversionWeight === 1 ? (0 <= fuel && fuel <= 80.0) : (0 <= fuel && fuel <= lbsUpperLimit)`

Also, this doesn't have to be done but, I think a cleaner alternative would is to make use of the already defined paramters in some methods that are referred to as `function(useLbs = false)` but replace with 
```js
randomFunction(useLbs = parseInt(NXDataStore.get("CONFIG_USING_METRIC_UNIT", "1")) === 1 ? false : true) {
//
}
```
Then add these parameter to method that don't already use it. The current method works fine though.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_FuelPredPage.js (hunk 1 above)
Comment (MisterChocker) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_FuelPredPage.js:138]: I don't understand your suggestion to 138

**Comment by MisterChocker** [comment]
> You need to amend all the range checks as well otherwise people who are using lbs will get a lot of "entry out of range" checks.

I don't think so, because I convert all entries before any checks.

**Comment by MisterChocker** [comment]
> make use of the already defined paramters in some methods

I tried, didn't worked.

**Comment by Lucky38i** [comment]
> > You need to amend all the range checks as well otherwise people who are using lbs will get a lot of "entry out of range" checks.
> 
> I don't think so, because I convert all entries before any checks.

I can't see where this._conversionWeight is instantiated or assigned (Maybe I'm not looking in the write place). From what I'm seeing are all values held in kg? but outputted in LBs ,if LBs is selected in the options menu, otherwise outputted in kg?)

**Comment by MisterChocker** [comment]
> From what I'm seeing are all values held in kg? but outputted in LBs ,if LBs is selected in the options menu, otherwise outputted in kg

yes, it will convert every input into KG, and convert output to the selected UNIT.

**Comment by Lucky38i** [comment]
> > From what I'm seeing are all values held in kg? but outputted in LBs ,if LBs is selected in the options menu, otherwise outputted in kg
> 
> yes, it will convert every input into KG, and convert output to the selected UNIT.

ah ok, disregard then. Only change is that conversion you did to percentages for some reason that I quoted beforehand.

**Review by wpine215 — APPROVED** [review]

**Comment by CaptainR3X** [comment]
Category: QA Tester
Name: GustavRex #4347
Date of testing: Current date 15/12/2020
Version of the sim: 1.11.7.0
PR Tested: #2419
Object of test: Test ability to change between KG and LBS

Testing technique: At gate, weight cross-calculations
Steps you've made to test the Pull Request:
1. Noting the KG´s displayed in-game
2. Calcultaing correct LBS
3. Noting the LBS in-game

Issues (If Any): Depends on how iffy you are with numbers :)
Overall Rating: Might need improvement? Idk how strict you are with the numbers 
Conclusions of the testing: Not entirely correct if you are as critical as I am :) 
image

**Comment by MisterChocker** [comment]
The difference is caused by taking a rounded value and converting it. The system converts a raw value and then rounds it.

**Comment by CaptainR3X** [comment]
Yeah just talked to DerL30N, he's looking at the negative values. Was not aware of the conversion b4

**Comment by CaptainR3X** [comment]
Upper ECAM display shows negative FOB value when empty after latest update

**Comment by CaptainR3X** [comment]
LGTM

**Review by Lucky38i — APPROVED** [review]
LGTM

**Comment by HugoTro** [comment]
Name: Hugo #3307
Date of testing: Current date 15/12/2020
Version of the sim: 1.11.7.0
PR Tested: #2419 

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Spawned in C&D
2. Turned batteries & ext pwr on
3. Checked the weights and converted them in google

Overall Rating: To be improved
Conclusions of the testing: I don't know why but the numbers seem rounded, for example at TO/GA power I had a fuel flow of 2400 KG/H and I converted it in pounds, it was showing 1088 lb and in the sim it was only 1080 lb...

**Comment by Highflyer2010** [comment]
Category: QA Tester
Name: Highflyer2010 #8932
Date of testing: Current date 15/12/2020
Version of the sim: 1.7.11.1
PR Tested: #2419 
Object of test: Add option to switch between KG and LBS

Testing technique: downloaded, started sim and switching the option from KG to LBS
Steps you've made to test the Pull Request:
1. spawned at TNCM - C & D
2. 9440kg fuel loaded FOB
3. switched to LBS now says 20820LBS FOB
4. switch back to KG. GW is 59600kg
5. switch lbs. 131400lbs GW
6. 110.8lbs in ZFW
7. switched to kg and i got 50.3lbs

Flight notes (If required): N/A

Issues (If Any):  UNTITLED page came up on the MCDU when i pressed "FUEL PRED"

Overall Rating: Good
Conclusions of the testing: LBS to KGS conversion are quite good i ended up with a (UNTITLED) Page while clicking onto to the "FUEL PRED" Page on the MCDU

Screenshot 2020-12-15 175408

**Comment by MisterChocker** [comment]
> Conclusions of the testing: I don't know why but the numbers seem rounded, for example at TO/GA power I had a fuel flow of 2400 KG/H and I converted it in pounds, it was showing 1088 lb and in the sim it was only 1080 lb...

yes displayed numbers are rounded. (ever 20)

**Comment by Highflyer2010** [comment]
> 
> 
> > 7. switched to kg and i got 50.3lbs
> 
> that sounds wrong tome me. you mean kg, right?

yes sorry i've edited and changed it

**Comment by MisterChocker** [comment]
> i ended up with a (UNTITLED) Page while clicking onto to the "FUEL PRED" Page on the MCDU

what have you done to get the untitled page?
What data did you inserted, in which order?

**Comment by Highflyer2010** [comment]
> 
> 
> > i ended up with a (UNTITLED) Page while clicking onto to the "FUEL PRED" Page on the MCDU
> 
> what have you done to get the untitled page?
> What data did you inserted, in which order?

i only double clicked on ZFW/ZFWCG   LSK could be why

**Comment by MisterChocker** [comment]
> i only double clicked on ZFW/ZFWCG LSK could be why

have you set an destination?

**Comment by MisterChocker** [comment]
Okay, I've found a bug in the original code. If no destination set there is an error..

**Comment by Highflyer2010** [comment]
> 
> 
> Okay, I've found a bug in the original code. If no destination set there is an error..

yeah i've just done a fake flight plan and all is good now.

**Comment by MisterChocker** [comment]
I've tested the fix for the error on fuel pred and it works just fine.

**Pull request merged**